### PR TITLE
feat: [code-213]: add useGenerate token hook

### PR DESCRIPTION
**Pull request by cjlee01**
### Summary

Created a hook to generate api token for usage with harness for git scm 

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by cjlee01** [comment]
retrigger titlecheck

File: src/modules/10-common/navigation/MainNav/MainNav.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export default function L1Nav(): React.ReactElement {
     NEW_LEFT_NAVBAR_SETTINGS &&
       document.getElementsByClassName(css.active)[0]?.scrollIntoView({ inline: 'nearest', block: 'nearest' })
   })
-
+  console.log(paths.toUser(params))
```
[the remark this answers is not in the document]
Comment (cjlee01) [reply src/modules/10-common/navigation/MainNav/MainNav.tsx:60]: removed

File: configs/devServerProxy.config.js [hunk 2]
```diff
@@ -110,6 +110,10 @@ module.exports = {
     pathRewrite: { '^/pm': '' },
     target: process.env.OPA_GOVERNANCE_UI_URL || 'http://localhost:3000'
   },
+  '/code-search': {
```
[the remark this answers is not in the document]
Comment (cjlee01) [reply configs/devServerProxy.config.js:113]: yeah will remove it @tan-nhu

File: src/modules/60-code/CodeApp.tsx [hunk 3]
```diff
@@ -93,6 +97,38 @@ const CODERemoteComponentMounter: React.FC<{
...
+        identifier: tokenName,
+        name: tokenName,
+        parentIdentifier: parentId || ''
+      })
+        .then(res => {
+          setToken(res)
+          return res
+        })
+        .catch(err => {
+          setToken(err)
```
Comment (tan-nhu) [inline src/modules/60-code/CodeApp.tsx:125]: Show a toaster here

**Review by tan-nhu — CHANGES_REQUESTED** [review]

**Review by sahithikolichala — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Pull request merged**